File: ui/src/app/services/comment.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { Comment } from "../Models/Comment";
import { Observable, Subject } from "rxjs";

const COMMENT_API = "https://flar-e.azurewebsites.net/api/Comments/";

@Injectable({
  providedIn: "root",
})
export class CommentService {
  public commentAdded: Subject<boolean>;

  constructor(private http: HttpClient) {
    this.commentAdded = new Subject<boolean>();
  }

  createComment(comment: Partial<Comment>): Observable<Comment> {
    return this.http.post<Comment>(COMMENT_API + "create", comment);
  }
  getComments(postId: number): Observable<Comment[]> {
    return this.http.get<Comment[]>(COMMENT_API + postId);
  }
}

File: BusinessLogicLayer/CommentService.cs
using Models;
using DataAccessLayer;

namespace BusinessLogicLayer

{
    public interface ICommentService
    {
        List<Comment> GetPostComments(int postId);
        Comment create_comment(Comment comment);
    }

    public class CommentService : ICommentService
    {
        private readonly ICommentsRepository commentRepository;

        public CommentService(ICommentsRepository commentRepository)
        {
            this.commentRepository = commentRepository;
        }

        public Comment create_comment(Comment comment)
        {
            return commentRepository.CreateComment(comment);
        }
        public List<Comment> GetPostComments(int postId)
        {
            return commentRepository.GetPostComments(postId);
        }
    }
}
